### PR TITLE
Fix tryStaticReshape to look at expanded extents. 

### LINKED ISSUE #1116: fd.ops.reshape seems to get the wrong input shape.
Reproducer:

```
import torch
from nvfuser import FusionDefinition

def test_expand_reshape():
    def expand_reshape() -> FusionDefinition:
        with FusionDefinition() as fd:
            x = fd.define_tensor([4, 5], [True, True])
            y = fd.ops.broadcast_in_dim(x, [4, 5, 6], [0, 1])
            y = fd.ops.reshape(y, [40, 3])
            fd.add_output(y)
        return fd

    fd = expand_reshape()

    x = torch.rand(4, 5, device="cuda")
    y = fd.execute([x])[0]
    print(y.shape)
```

```
$ pytest test.py -s
============================================================================================= test session starts =============================================================================================
platform linux -- Python 3.10.11, pytest-7.4.2, pluggy-1.3.0
rootdir: /opt/pytorch/nvfuser
plugins: xdist-3.3.1, hypothesis-6.88.1
collected 1 item

test.py F

================================================================================================== FAILURES ===================================================================================================
_____________________________________________________________________________________________ test_expand_reshape _____________________________________________________________________________________________

    def test_expand_reshape():
        def expand_reshape() -> FusionDefinition:
            with FusionDefinition() as fd:
                x = fd.define_tensor([4, 5], [True, True])
                y = fd.ops.broadcast_in_dim(x, [4, 5, 6], [0, 1])
                y = fd.ops.reshape(y, [40, 3])
                fd.add_output(y)
            return fd

>       fd = expand_reshape()

test.py:13:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test.py:6: in expand_reshape
    with FusionDefinition() as fd:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self =
def nvfuser_fusion_id0(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(shape=[4, 5], contiguity=[True, Tru...  V8 = fd.define_vector([S6, S7], dtype=DataType.Int)
    T9 = fd.ops.reshape(T5, new_shape=V8)
    fd.add_output(T9)


type = None, value = None, traceback = None

    def __exit__(self, type, value, traceback):
>       self._finalize_definition()
E       RuntimeError: original_num_elements == new_num_elements INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/transform_view.cpp":372, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. Total element counts across view operation must match: 20 vs 120
E       Exception raised from AnalyzeViewTransformation at /opt/pytorch/nvfuser/csrc/transform_view.cpp:372 (most recent call first):
E       frame #0: nvfuser::nvfCheckFail(char const*, char const*, unsigned int, std::string const&) + 0x8d (0x7f888d9e37c7 in /usr/local/lib/python3.10/site-packages/torch/lib/libnvfuser_codegen.so)
E       frame #1: nvfuser::nvfErrorFail(char const*, char const*, unsigned int, char const*, std::string const&) + 0x53 (0x7f888db25f43 in /usr/local/lib/python3.10/site-packages/torch/lib/libnvfuser_codegen.so)
E       frame #2: nvfuser::analyzeView(nvfuser::TensorView const*, std::vector<long, std::allocator<long> > const&, std::vector<long, std::allocator<long> > const&) + 0x10a2 (0x7f888df68392 in /usr/local/lib/python3.10/site-packages/torch/lib/libnvfuser_codegen.so)
E       frame #3: nvfuser::reshape(nvfuser::TensorView*, std::vector<long, std::allocator<long> > const&, std::vector<long, std::allocator<long> > const&) + 0xab (0x7f888ddc931b in /usr/local/lib/python3.10/site-packages/torch/lib/libnvfuser_codegen.so)
E       frame #4: nvfuser::reshape(nvfuser::TensorView*, std::vector<nvfuser::Val*, std::allocator<nvfuser::Val*> > const&) + 0x3f0 (0x7f888ddcae30 in /usr/local/lib/python3.10/site-packages/torch/lib/libnvfuser_codegen.so)
E       frame #5: nvfuser::python_frontend::ReshapeOpRecord::operator()(nvfuser::python_frontend::FusionState&) + 0x4e (0x7f88800b30fe in /opt/pytorch/nvfuser/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
E       frame #6: nvfuser::python_frontend::FusionState::buildFusionIr(nvfuser::Fusion*) + 0xac (0x7f888df9e55c in /usr/local/lib/python3.10/site-packages/torch/lib/libnvfuser_codegen.so)
E       frame #7: nvfuser::python_frontend::FusionDefinition::finalizeDefinition() + 0x114 (0x7f888df99f04 in /usr/local/lib/python3.10/site-packages/torch/lib/libnvfuser_codegen.so)
E       frame #8: <unknown function> + 0xdd237 (0x7f888005d237 in /opt/pytorch/nvfuser/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
E       frame #9: <unknown function> + 0x16d927 (0x7f88800ed927 in /opt/pytorch/nvfuser/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
E       frame #10: <unknown function> + 0x14fcbb (0x56011c719cbb in /usr/local/bin/python3.10)
E       frame #11: _PyObject_MakeTpCall + 0x2ab (0x56011c7136ab in /usr/local/bin/python3.10)
E       frame #12: <unknown function> + 0x15b60f (0x56011c72560f in /usr/local/bin/python3.10)
E       frame #13: _PyEval_EvalFrameDefault + 0x5165 (0x56011c70e245 in /usr/local/bin/python3.10)
E       frame #14: <unknown function> + 0x15b315 (0x56011c725315 in /usr/local/bin/python3.10)
E       frame #15: _PyEval_EvalFrameDefault + 0x345 (0x56011c709425 in /usr/local/bin/python3.10)
E       frame #16: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #17: _PyEval_EvalFrameDefault + 0x345 (0x56011c709425 in /usr/local/bin/python3.10)
E       frame #18: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #19: _PyEval_EvalFrameDefault + 0x2e8c (0x56011c70bf6c in /usr/local/bin/python3.10)
E       frame #20: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #21: _PyEval_EvalFrameDefault + 0x2e8c (0x56011c70bf6c in /usr/local/bin/python3.10)
E       frame #22: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #23: _PyEval_EvalFrameDefault + 0x5165 (0x56011c70e245 in /usr/local/bin/python3.10)
E       frame #24: <unknown function> + 0x15b315 (0x56011c725315 in /usr/local/bin/python3.10)
E       frame #25: _PyEval_EvalFrameDefault + 0x5165 (0x56011c70e245 in /usr/local/bin/python3.10)
E       frame #26: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #27: _PyObject_FastCallDictTstate + 0x18d (0x56011c712a0d in /usr/local/bin/python3.10)
E       frame #28: _PyObject_Call_Prepend + 0x6c (0x56011c722e0c in /usr/local/bin/python3.10)
E       frame #29: <unknown function> + 0x2282fd (0x56011c7f22fd in /usr/local/bin/python3.10)
E       frame #30: _PyObject_MakeTpCall + 0x2ab (0x56011c7136ab in /usr/local/bin/python3.10)
E       frame #31: _PyEval_EvalFrameDefault + 0x5d55 (0x56011c70ee35 in /usr/local/bin/python3.10)
E       frame #32: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #33: _PyEval_EvalFrameDefault + 0x6fe (0x56011c7097de in /usr/local/bin/python3.10)
E       frame #34: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #35: _PyEval_EvalFrameDefault + 0x2e8c (0x56011c70bf6c in /usr/local/bin/python3.10)
E       frame #36: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #37: _PyEval_EvalFrameDefault + 0x5165 (0x56011c70e245 in /usr/local/bin/python3.10)
E       frame #38: <unknown function> + 0x15b315 (0x56011c725315 in /usr/local/bin/python3.10)
E       frame #39: _PyEval_EvalFrameDefault + 0x5165 (0x56011c70e245 in /usr/local/bin/python3.10)
E       frame #40: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #41: _PyObject_FastCallDictTstate + 0x18d (0x56011c712a0d in /usr/local/bin/python3.10)
E       frame #42: _PyObject_Call_Prepend + 0x6c (0x56011c722e0c in /usr/local/bin/python3.10)
E       frame #43: <unknown function> + 0x2282fd (0x56011c7f22fd in /usr/local/bin/python3.10)
E       frame #44: PyObject_Call + 0xe2 (0x56011c725e92 in /usr/local/bin/python3.10)
E       frame #45: _PyEval_EvalFrameDefault + 0x2e8c (0x56011c70bf6c in /usr/local/bin/python3.10)
E       frame #46: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #47: _PyEval_EvalFrameDefault + 0x345 (0x56011c709425 in /usr/local/bin/python3.10)
E       frame #48: <unknown function> + 0x15b315 (0x56011c725315 in /usr/local/bin/python3.10)
E       frame #49: _PyEval_EvalFrameDefault + 0x149c (0x56011c70a57c in /usr/local/bin/python3.10)
E       frame #50: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #51: _PyEval_EvalFrameDefault + 0x2e8c (0x56011c70bf6c in /usr/local/bin/python3.10)
E       frame #52: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #53: _PyEval_EvalFrameDefault + 0x345 (0x56011c709425 in /usr/local/bin/python3.10)
E       frame #54: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #55: _PyEval_EvalFrameDefault + 0x149c (0x56011c70a57c in /usr/local/bin/python3.10)
E       frame #56: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #57: _PyEval_EvalFrameDefault + 0x2e8c (0x56011c70bf6c in /usr/local/bin/python3.10)
E       frame #58: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #59: _PyEval_EvalFrameDefault + 0x5165 (0x56011c70e245 in /usr/local/bin/python3.10)
E       frame #60: <unknown function> + 0x15b315 (0x56011c725315 in /usr/local/bin/python3.10)
E       frame #61: _PyEval_EvalFrameDefault + 0x5165 (0x56011c70e245 in /usr/local/bin/python3.10)
E       frame #62: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #63: _PyObject_FastCallDictTstate + 0x18d (0x56011c712a0d in /usr/local/bin/python3.10)

nvfuser/__init__.py:41: RuntimeError
=========================================================================================== short test summary info ===========================================================================================
FAILED test.py::test_expand_reshape - RuntimeError: original_num_elements == new_num_elements INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/transform_view.cpp":372, please report a bug with repro script to NVFuser at https://github.c...
============================================================================================== 1 failed in 1.27s ==============================================================================================
```

For some reason, nvFuser thinks `original_num_elements` is 20, which is wrong. I suspect it's a collateral damage by #1043 .

### PR DESCRIPTION
**Pull request by wujingyue**
Fixes #1116

### CODE REVIEW

**Comment by wujingyue** [comment]
@protonu Sorry, I'm adding more uses of `evaluateInt` that your #1104 plans to get rid of...

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
@xwang233 do you see https://gitlab-master.nvidia.com/dl/pytorch/fuser-gh-mirror/-/jobs/72134299 a concern for this PR? I'm not sure why it was failing.

**Comment by xwang233** [comment]
Err, sorry for the confusion. I've fixed the CI script then. It's not your fault. Feel free to restart a new pipeline if that's a concern.

File: csrc/ops/alias.cpp [hunk 1]
```diff
@@ -77,21 +77,21 @@ TensorView* tryStaticReshape(
     const std::vector<Val*>& new_sizes) {
   std::vector<int64_t> inp_sizes(inp_dom.size());
   for (const auto i : c10::irange(inp_dom.size())) {
-    auto id = inp_dom.at(i);
-    auto id_size = id->extent()->getInt();
-    if (!id_size.has_value()) {
+    IterDomain* id = inp_dom[i];
+    Val* id_size = id->getMaybeExpandedExtent();
```
Comment (jjsjann123) [inline csrc/ops/alias.cpp:81]: Looks like this is the only logic change in this PR?

**Review by jjsjann123 — APPROVED** [review]
LGTM

File: csrc/ops/alias.cpp (hunk 1 above)
Comment (wujingyue) [reply csrc/ops/alias.cpp:81]: No. Also the use of `evaluateInt`. For expanded broadcast dimensions, there's an extra indirection that `getInt` or `value` doesn't just work.

Comment (jjsjann123) [reply csrc/ops/alias.cpp:81]: Got ya. Thanks for explaining.

**Pull request merged**